src/services/pathfinding.service.ts

```diff
@@ -14,20 +25,221 @@ class PathFindingService {
+      start,
+      escalatorDirection
+    );
+    const endFloorConnectors: ConnectorPOI[] = this.findReachableConnectorsFromPOI(
+      end,
+      escalatorDirection
+    );
+
+    const pathsFromStartToConnectors = this.getPathsBetweenPOIAndConnectors(
+      start,
+      startFloorConnectors,
+      startFloorNodes
+    );
+    const pathsFromEndToConnectors = this.getPathsBetweenPOIAndConnectors(
+      end,
+      endFloorConnectors,
+      endFloorNodes
+    );
+
+    return this.getShortestPathThroughBuilding(
+      pathsFromStartToConnectors,
+      pathsFromEndToConnectors
+    );
+  };
+
+  /**
+   * Returns all connector POI's which are in the same floor and building as the given POI.
+   *
+   * @param poi POI whose building/floor is searched for conenctors
+   * @param escalatorDirection Accepted escalator direction for current travel plan
+   *
+   * @returns A list of connectorPOI's
+   */
+  private findReachableConnectorsFromPOI = (
+    poi: POI,
+    escalatorDirection: POICategory
+  ): ConnectorPOI[] => {
+    const connectors = POIInfo.filter(
+      ({ category, level, buildingId }) =>
+        poi.level === level &&
+        poi.buildingId === buildingId &&
+        (category === POICategory.Exit ||
+          category === POICategory.Stairs ||
+          category === escalatorDirection ||
+          category === POICategory.Elevator)
+    ) as ConnectorPOI[];
+    return connectors;
+  };
+
+  /**
+   * Returns the path distance
+   *
+   * @param path path for which to calculate distance
+   *
+   * @returns The distance traversed by the path
+   */
+  private getPathDistance = (path: Line[]): number => {
+    let distance = 0;
+    path.forEach((line) => {
+      distance += getDistance(line[0], line[1]);
+    });
+    return distance;
+  };
+
+  /**
+   * Returns a list of valid paths between a POI and all the other connectors on the same building/floor
+   *
+   * @param poi POI which serves as the start point for per-floor navigation
+   * @param connectors List of reachable connectors. These POI's must be on the
+   * same building/floor as the starting POI
+   * @param nodes Travel nodes for the given floor
+   *
+   * @returns List of valid paths
+   */
+  private getPathsBetweenPOIAndConnectors = (
+    poi: POI,
+    connectors: ConnectorPOI[],
+    nodes: TravelNode[]
+  ): FloorPath[] => {
+    const pathsToConnectors: FloorPath[] = [];
+
+    connectors.forEach((connector) => {
+      pathsToConnectors.push({
+        buildingId: connector.buildingId,
+        level: connector.level,
+        connectorType: connector.category,
+        path: this.findPathOnFloor(nodes, poi.location, connector.location),
+      });
+    });
+
+    return pathsToConnectors;
+  };
+
+  /**
+   * Searches through valid paths between start and end POI to find the shortest one.
+   *
+   * @param startFloorPathsToConnectors
+   * @param endFloorPathsToConnectors
```

Comment:
```
   * @param startFloorPathsToConnectors
   * @param endFloorPathsToConnectors
```
These may need more descriptions
Might also just double check for other params in the file that don't have descriptions on them, especially since this file is quite huge

Reply:
Yes, I will add descriptions to the parameters of the functions in this file.